core/src/data/dataSource.h

```diff
@@ -128,13 +132,10 @@ class DataSource : public std::enable_shared_from_this<DataSource> {
     // Generation of dynamic DataSource state (incremented for each update)
     int64_t m_generation = 1;
 
-    // URL template for requesting tiles from a network or filesystem
-    std::string m_urlTemplate;
-
-    std::unique_ptr<RawCache> m_cache;
-
     /* vector of raster sources (as raster samplers) referenced by this datasource */
     std::vector<std::shared_ptr<DataSource>> m_rasterSources;
+
+    std::unique_ptr<RawDataSource> m_sources;
```

Comment:
Yes, I'm also not pleased with the `RawDataSource` and `-DataSource` naming. I thought of renaming `DataSource` to `TileDataSource` since the result of `DataSource::parse` is a `TileData` object. Still open for suggestions and thinking about further refactoring of DataSource